kura/org.eclipse.kura.linux.bluetooth/src/main/java/org/eclipse/kura/linux/bluetooth/le/beacon/BluetoothAdvertisingData.java

```diff
@@ -43,10 +43,12 @@ public static String getData(String uuid, Integer major, Integer minor, String c
             txPowerString = Integer.toHexString(txPower);
             if (txPowerString.length() == 1) {
                 txPowerString = "0" + txPowerString;
+            } else if (txPowerString.length() > 2) {
+                txPowerString = txPowerString.substring(txPowerString.length() - 2);
             }
         } else {
             txPowerString = Integer.toHexString(txPower);
-            txPowerString = txPowerString.substring(txPowerString.length() - 2, txPowerString.length());
+            txPowerString = txPowerString.substring(txPowerString.length() - 2);
         }
 
```

Comment:
I think that this approximation is a bit misleading. If we limit the txPower between -127 and 126 and we receive a value outside this range, we should cut the value to the minimum/maximum value (i.e. 200 -> 126 or -200 -> -127).